### DISCUSSION TITLE
crawlFile怎么搞都搞不定....

### DISCUSSION THREAD

**Discussion by winner106** [coder-hxl/x-crawl#35] [General]
```js
	await myXCrawl.crawlFile({
		url: 'https://query1.finance.yahoo.com/v7/finance/download/JD?period1=1683336000&period2=1683936000&interval=1d&events=history&includeAdjustedClose=true',
		storeDir: './public/data/financial',
		fileName: 'test',
	});
```
得到的结果是这样的`test.csv;charset=utf-8`, 请教大神指引, 以上接口需要开梯子, 直接可下载文件的.

**Comment by coder-hxl** [comment]
将 `extension` 选项设置为  `.csv`

**Reply by winner106** [reply]
搞定了, 非常感谢. 再请问下, 使用` await myXCrawl.crawlFile({ targets, storeDir: './upload' })`去批量下载文件时, 是不是无法对每个文件命名? 我目前是一个个去`for of`循环下载的. 如果fileName能支持传入函数就可以解决了

**Reply by coder-hxl** [reply]
可以针对每个文件进行命名，修改 [CrawlFileDetailTargetConfig](https://github.com/coder-hxl/x-crawl/blob/main/docs/cn.md#CrawlFileDetailTargetConfig) 的 `fileName` 配置选项，需要提前配置好。
感觉直接传字符串更加方便点，如果是函数的话，在爬每个目标前还得回调一下。

**Reply by winner106** [reply]
这样会不会更方便? 既支持直接传入字符串, 也支持传入数组
```js
const CODES = ['NTES', 'BIDU', 'JD'];
await myXCrawl.crawlFile({
  url: CODES.map((d) => `https://query1.finance.yahoo.com/v7/finance/download/${d}`),
  storeDir: './public/data/financial',
  fileName: CODES,
  extension: '.csv',
});
```

**Reply by coder-hxl** [reply]
这种方式也不错。你可以去 https://github.com/coder-hxl/x-crawl/issues/new/choose 提一下新想法建议 😄

**Reply by winner106** [reply]
ok
